### DISCUSSION TITLE
Call to a member function getForeignKeyName() on null

### DISCUSSION THREAD

**Discussion by ousid** [filamentphp/filament#12100] [Help]
### Package

Form builder

### Package Version

v3.2.60

### How can we help you?

I'm trying to build a form with relations, using the __Form Builder__, but once I filled the form, and try to submit it, it looks the relations didn't get created, and give me an error

### Error
`
Call to a member function getForeignKeyName() on null
`

### Steps to produce
- Build the form
- Attach the relationships
- Submit the form

### Code Sample

- **Select Component**
```php
Forms\Components\Select::make('manufacturer_id')
    ->label('Manufacturer')
    ->relationship(
        'manufacturer',
        'name',
        $this->whereActiveRecords()
    )
    ->required()
    ->preload()
    ->native(false)
    ->searchable()
    ->createOptionForm(
        fn (): array => $this->manufacturerForm()
    )
    ->createOptionAction($this->handleExtraFields()),
```

- **Model Relation**
```php
    public function manufacturer(): BelongsTo
    {
        return $this->belongsTo(Manufacturer::class, 'manufacturer_id');
    }
```

- **Store Method**
```php
public function store()
{
  // ...
  $product = Product::create($this->form->getState());
 // the product will be stored normally.
  $this->form->model($product)->saveRelationships();
  // ...
}
```

__PS__: Tried to investigate, through the source code, and through the web, but didn't find any useful resource.

Any help would be much appreciated it.

cc, @danharrin

**Comment by yuvrajtimalsina** [comment]
Have you found the solution to it?
